### PR TITLE
♻️Refactor new loaders code.

### PR DESCRIPTION
**Pull request by sparhami**
- Move some presentation logic from JS into CSS
- Change custom logos to define their own SVGs
  * This allows the logos to specify their own viewbox
  * The logos still need to be positioned relative to the overall svg dimensions (centered and sized to fit within the spinner arc), this can be changed using a `<foreignObject>` and put the svg in there
- Make API for custom loaders specifying a color more explicit (rather
than just reading it off the custom loader Element)
- Change coordinate systems for the logos to start at zero so they are easier
  to reason about.
- Fix the default loader size, which defaults to 300x300 on Firefox when
no size is specified.
- Fix the smaller loader being the wrong size, due to the SVG being
sized down. This may overflow the parent container, but should position
correctly as it is centered.

### CODE REVIEW

File: extensions/amp-loader/0.1/amp-loader.js [hunk 1]
```diff
@@ -292,28 +218,34 @@ class LoaderBuilder {
...
-      svgNode.setAttribute('fill', color);
-      return svgNode;
+      return {
+        content,
+      };
     }
-
-    const customLogo = this.element_.createLoaderLogo();
-    return customLogo || null;
+    return this.element_.createLoaderLogo() || {};
```
Comment (wassgha) [inline extensions/amp-loader/0.1/amp-loader.js]: Let's make this return an empty object by default

Comment (sparhami) [reply extensions/amp-loader/0.1/amp-loader.js]: Done.

File: src/base-element.js [hunk 2]
```diff
@@ -437,13 +437,13 @@ export class BaseElement {
   /**
    * Subclasses can override this method to provide a svg logo that will be
    * displayed as the loader.
-   * @return {?{
+   * @return {!{
```
Comment (wassgha) [inline src/base-element.js]: This is unnecessary as objects are non-nullable by default

File: src/custom-element.js [hunk 3]
```diff
@@ -1032,7 +1032,7 @@ function createBaseCustomElementClass(win) {
 
     /**
      * Creates a loader logo.
-     * @return {?{
+     * @return {!{
```
Comment (wassgha) [inline src/custom-element.js]: ditto

**Pull request merged**